File: csharp/core/UnivecError.cs
// UnivecError - the SDK error type. Carries the pipeline error code,
// the originating context and cleaned result/spec snapshots.

namespace UnivecSdk;

public class UnivecError : Exception
{
    public bool IsUnivecError = true;
    public string Sdk = "Univec";
    public string Code;
    public Context? Ctx;
    public object? ResultVal;
    public object? SpecVal;

    public UnivecError(string code, string msg, Context? ctx)
        : base(msg)
    {
        Code = code;
        Ctx = ctx;
    }
}

File: swift/Sources/UnivecSdk/core/UnivecError.swift
// UnivecError - the SDK error type. Carries the pipeline error code, the
// originating context and cleaned result/spec snapshots.

import Foundation

public final class UnivecError: Error, CustomStringConvertible {
  public let isUnivecError = true
  public let sdk = "Univec"
  public let code: String
  public let message: String
  public var ctx: Context?
  public var resultVal: Value = .noval
  public var specVal: Value = .noval

  public init(_ code: String, _ message: String, _ ctx: Context?) {
    self.code = code
    self.message = message
    self.ctx = ctx
  }

  public var description: String { message }
}
